spec/compilers/array_access

```diff
@@ -0,0 +1,23 @@
+module Test {
+  fun test : Maybe(String) {
+    [
+      "Hello",
+      "Blah",
+      "Joe"
+    ][1]
+  }
+
+  fun test1 : Maybe(a) {
+    [][1]
+  }
+}
+--------------------------------------------------------------------------------
+const $Test = new(class {
+  test() {
+    return [`Hello`, `Blah`, `Joe`][1]
```

Comment:
The compiler needs to return an actual `Maybe` type from the runtime https://github.com/mint-lang/mint-runtime/blob/master/source/Maybe.js as I though it should be something like ```return _at([`Hello`, `Blah`, `Joe`], 1)``` where the `_at` is implemented in mint-runtime and unwrapped here: https://github.com/mint-lang/mint/blob/master/src/compilers/top_level.cr#L124

In this specific case we can do even better: ```return new Just(`Hello`)```